src/app/modules/auth/sign-up/sign-up.component.ts

```diff
@@ -18,123 +67,199 @@ import { ResendEmailConfirmationMutation } from '../auth.graphql';
+  submitEmailSignup(form: NgForm) {
+    if (!form.valid) return;
+    this.loadingEmail = true;
+    this.emailError = '';
+    this.apollo.mutate({
+      mutation: AUTH_LOGIN_MUTATION,
+      variables: { email: this.emailForOtp }
+    }).subscribe(
+      (res: any) => {
+        this.loadingEmail = false;
+        if (res?.data?.authLogin) {
+          this.otpStep = true;
+          this.emailStep = false;
+          this.otpError = '';
+        } else {
+          this.emailError = 'Unexpected error. Please try again.';
         }
-      }, (err: any) => this.loadingCities = false);
+      },
+      err => {
+        this.loadingEmail = false;
+        this.emailError = err?.message || 'Failed to send OTP. Please try again.';
+      }
+    );
   }
 
-  onSearch(query: any) {
-    if (!query.term) {
-      query = null;
-      return;
-    }
-    this.searchEmitter.emit(query);
+  submitOtp(form: NgForm) {
+    if (!form.valid) return;
+    this.loadingOtp = true;
+    this.otpError = '';
+    this.apollo.mutate({
+      mutation: VERIFY_OTP_MUTATION,
+      variables: { email: this.emailForOtp, otp: this.otp }
+    }).subscribe(
+      (res: any) => {
+        this.loadingOtp = false;
+        const tokenObj = res?.data?.verifyOtp;
+        if (tokenObj?.accessToken) {
+          this.tokenService.storeToken(tokenObj);
+          this.fetchAndCheckUserDetails();
+        } else {
+          this.otpError = 'Invalid OTP. Please try again.';
+        }
+      },
+      err => {
+        this.loadingOtp = false;
+        this.otpError = err?.message || 'Failed to verify OTP. Please try again.';
+      }
+    );
   }
 
-  searchCity(name: string) {
-    if (name.trim()) {
-      return this.apollo.query({
-          query: CitiesSearchQuery,
-          variables: {
-            q: name,
-            type: 'city'
-          }
-        })
-        .pipe(
-          map((i: any) => i.data.locationAutocomplete),
-          tap(() => this.loadingCities = false)
-        );
-    }
+  fetchAndCheckUserDetails() {
+    this.apollo.query({
+      query: USER_CURRENT_QUERY,
+      fetchPolicy: 'network-only'
+    }).subscribe(
+      (result: any) => {
+        const user = result?.data?.userCurrent;
+        if (
+          !user ||
+          !user.firstName ||
+          !user.email ||
+          !user.city ||
+          !user.city.id
+        ) {
+          this.userDetailsForm = {
+            firstName: user?.firstName || '',
+            cityId: user?.city?.id || null,
+            email: user?.email || this.emailForOtp
+          };
+          this.showUserDetailsModal = true;
+          this.loadUserDetailsCities();
+        } else {
+          this.onSignUp();
+        }
+      },
+      err => {
+        this.userDetailsForm = {
+          firstName: '',
+          cityId: null,
+          email: this.emailForOtp
+        };
+        this.showUserDetailsModal = true;
+        this.loadUserDetailsCities();
+      }
+    );
```

Comment:
The same OTP logic is being repeated in multiple files. Create a re-usable component to reduce code duplication